translator/translate/otel/receiver/otlp/translator.go

```diff
@@ -73,6 +74,97 @@ func (t *translator) ID() component.ID {
 	return component.NewIDWithName(t.factory.Type(), t.Name())
 }
 
+var (
+	registryMu sync.RWMutex
+	registry   = make(map[string]common.ComponentTranslator)
+	counter    int
+)
+
+func getEndpointKey(conf *confmap.Conf, configKey string, index int, signal pipeline.Signal) (string, error) {
+	grpcEndpoint := defaultGrpcEndpoint
+	httpEndpoint := defaultHttpEndpoint
+	tlsCert, tlsKey := "", ""
+
+	// Handle application signals defaults
+	if appSignalsConfigKeys, ok := common.AppSignalsConfigKeys[signal]; ok {
+		for _, appKey := range appSignalsConfigKeys {
+			if configKey == appKey {
+				grpcEndpoint = defaultAppSignalsGrpcEndpoint
+				httpEndpoint = defaultAppSignalsHttpEndpoint
+				break
+			}
+		}
+	}
+
+	if conf != nil && conf.IsSet(configKey) {
+		otlpMap := common.GetIndexedMap(conf, configKey, index)
+		if grpc, ok := otlpMap["grpc_endpoint"]; ok {
+			grpcEndpoint = grpc.(string)
+		}
+		if http, ok := otlpMap["http_endpoint"]; ok {
+			httpEndpoint = http.(string)
+		}
+		if tls, ok := otlpMap["tls"].(map[string]interface{}); ok {
+			if cert, ok := tls["cert_file"].(string); ok {
+				tlsCert = cert
+			}
+			if key, ok := tls["key_file"].(string); ok {
+				tlsKey = key
+			}
+		}
+	}
+
+	portKey := fmt.Sprintf("%s|%s", grpcEndpoint, httpEndpoint)
+	fullKey := fmt.Sprintf("%s|%s|%s", portKey, tlsCert, tlsKey)
```

Comment:
Rather than a string key, would it make sense to have a struct with string fields? It's still comparable, but would make accessing the fields easier. We're basically doing all of the translation parsing logic here. Maybe we could just pass this struct in when creating the translator. That way we don't have to do it again.